pallets/ocex/src/lib.rs

```diff
@@ -0,0 +1,1021 @@
+						.proxies
+						.iter()
+						.position(|account| *account == proxy)
+						.ok_or(Error::<T>::ProxyNotFound)?;
+					account_info.proxies.remove(proxy_positon);
+					<IngressMessages<T>>::mutate(|ingress_messages| {
+						ingress_messages.push(
+							polkadex_primitives::ingress::IngressMessages::RemoveProxy(
+								main_account.clone(),
+								proxy.clone(),
+							),
+						);
+					});
+				}
+				Self::deposit_event(Event::ProxyRemoved { main: main_account.clone(), proxy });
+				Ok(())
+			})
+		}
+
+		/// Extrinsic used by enclave to submit balance snapshot and withdrawal requests
+		#[pallet::weight((590_500_000 as Weight).saturating_add(T::DbWeight::get().reads(3 as Weight)).saturating_add(T::DbWeight::get().writes(5 as Weight)))]
+		pub fn submit_snapshot(
+			origin: OriginFor<T>,
+			mut snapshot: EnclaveSnapshot<
+				T::AccountId,
+				WithdrawalLimit,
+				AssetsLimit,
+				SnapshotAccLimit,
+			>,
+			signature: T::Signature,
+		) -> DispatchResult {
+			let enclave = ensure_signed(origin)?;
+			ensure!(
+				<RegisteredEnclaves<T>>::contains_key(&enclave),
+				Error::<T>::SenderIsNotAttestedEnclave
+			);
+
+			let last_snapshot_serial_number =
+				if let Some(last_snapshot_number) = <SnapshotNonce<T>>::get() {
+					last_snapshot_number
+				} else {
+					0
+				};
+			ensure!(
+				snapshot.snapshot_number.eq(&(last_snapshot_serial_number + 1)),
+				Error::<T>::SnapshotNonceError
+			);
+			let bytes = snapshot.encode();
+			ensure!(
+				signature.verify(bytes.as_slice(), &enclave),
+				Error::<T>::EnclaveSignatureVerificationFailed
+			);
+			let current_snapshot_nonce = snapshot.snapshot_number;
+			if snapshot.withdrawals.keys().len() > 0 {
+				ensure!(
+					<OnChainEvents<T>>::try_mutate(|onchain_events| {
+						onchain_events.try_push(
+							polkadex_primitives::ocex::OnChainEvents::GetStorage(
+								polkadex_primitives::ocex::Pallet::OCEX,
+								polkadex_primitives::ocex::StorageItem::Withdrawal,
+								snapshot.snapshot_number,
+							),
+						)?;
+						Ok::<(), ()>(())
+					})
+					.is_ok(),
+					Error::<T>::OnchainEventsBoundedVecOverflow
+				);
+			}
+			<Withdrawals<T>>::insert(current_snapshot_nonce, snapshot.withdrawals.clone());
+			<FeesCollected<T>>::insert(current_snapshot_nonce, snapshot.fees.clone());
+			snapshot.withdrawals = Default::default();
+			snapshot.fees = Default::default();
+			<Snapshots<T>>::insert(current_snapshot_nonce, snapshot.clone());
+			<SnapshotNonce<T>>::put(current_snapshot_nonce);
+			Ok(())
+		}
+
+		// FIXME Only for testing will be removed before mainnet launch
+		/// Insert Enclave
+		#[doc(hidden)]
+		#[pallet::weight(10000 + T::DbWeight::get().writes(1))]
+		pub fn insert_enclave(origin: OriginFor<T>, encalve: T::AccountId) -> DispatchResult {
+			T::GovernanceOrigin::ensure_origin(origin)?;
+			let timestamp = <timestamp::Pallet<T>>::get();
+			<RegisteredEnclaves<T>>::insert(encalve, timestamp);
+			Ok(())
+		}
+
```

Comment:
I assume this should be removed?

Reply:
We can do this before the runtime upgrade